rosidl_generator_c/src/primitives_array_functions.c

```diff
@@ -0,0 +1,68 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "rosidl_generator_c/primitives_array_functions.h"
+
+#define ROSIDL_GENERATOR_C__DEFINE_PRIMITIVE_ARRAY_FUNCTIONS(STRUCT_NAME, TYPE_NAME) \
+  bool rosidl_generator_c__ ## STRUCT_NAME ## __Array__init( \
+    rosidl_generator_c__ ## STRUCT_NAME ## __Array * array, size_t size) \
+  { \
+    TYPE_NAME * data = NULL; \
+    if (size) { \
+      data = malloc(sizeof(TYPE_NAME) * size); \
+      if (!data) { \
+        return false; \
+      } \
+    } \
+    array->data = data; \
+    array->size = size; \
+    array->capacity = size; \
+    return true; \
+  } \
```

Comment:
Absolutely. I added it for all cases I found in 0694ccbeedec3eb5af33d483d1420a8e38b9ea8d